lxd/main_init.go

```diff
@@ -163,6 +166,51 @@ func (c *cmdInit) Run(cmd *cobra.Command, args []string) error {
 		config.Cluster.ClusterCertificate = string(content)
 	}
 
+	// Check if we got a cluster join token, if so, fill in the config with it.
+	if config.Cluster != nil && config.Cluster.ClusterToken != "" {
+		joinToken, err := clusterMemberJoinTokenDecode(config.Cluster.ClusterToken)
+		if err != nil {
+			return errors.Wrapf(err, "Invalid cluster join token")
+		}
+
+		// Set server name from join token
+		config.Cluster.ServerName = joinToken.ServerName
+
+		// Attempt to find a working cluster member to use for joining by retrieving the
+		// cluster certificate from each address in the join token until we succeed.
+		for _, clusterAddress := range joinToken.Addresses {
+			// Cluster URL
+			_, _, err := net.SplitHostPort(clusterAddress)
+			if err != nil {
+				clusterAddress = fmt.Sprintf("%s:%d", clusterAddress, shared.DefaultPort)
+			}
+			config.Cluster.ClusterAddress = clusterAddress
+
+			// Cluster certificate
+			cert, err := shared.GetRemoteCertificate(fmt.Sprintf("https://%s", config.Cluster.ClusterAddress), version.UserAgent)
+			if err != nil {
+				fmt.Printf("Error connecting to existing cluster node %q: %v\n", clusterAddress, err)
```

Comment:
Cluster member father than cluster node (at least that's my understanding that we don't use node terminology in user facing statements)

Reply:
Ah yeah, that's a copy/paste, I'll push a commit to tweak the rest.